cns/service.go

```diff
@@ -48,64 +48,65 @@ func NewService(name, version, channelMode string, store store.KeyValueStore) (*
 	}, nil
 }
 
-// GetAPIServerURL returns the API server URL.
-func (service *Service) getAPIServerURL() string {
-	urls, _ := service.GetOption(acn.OptCnsURL).(string)
-	if urls == "" {
-		urls = defaultAPIServerURL
+func (service *Service) AddListener(config *common.ServiceConfig) error {
+	// Fetch and parse the API server URL.
+	var nodeURL *url.URL
+	cnsURL, _ := service.GetOption(acn.OptCnsURL).(string)
```

Comment:
is there a reason why we are not checking for err here?

Reply:
GetOption does not return error

Reply:
@paulyufan2 The type assertion returns an `ok` value here. If that's `false` here, `cnsURL` will be the empty string which will lead to a silent misbehavior for a DRI to diagnose rather than an actionable error with immediate resolution.